src/Microsoft.OData.Edm/Csdl/CsdlXmlWriter.cs

```diff
@@ -96,8 +97,31 @@ private void WriteConceptualModelsElement()
 
         private void WriteReferenceElements()
         {
-            EdmModelReferenceElementsXmlVisitor visitor = new EdmModelReferenceElementsXmlVisitor(this.model, this.writer, this.edmxVersion);
-            visitor.VisitEdmReferences(this.model);
+            EdmModelReferenceElementsXmlVisitor visitor;
+            IEnumerable<IEdmReference> references = model.GetEdmReferences();
+            if (model != null && references != null)
+            {
+                foreach (IEdmReference reference in references)
+                {
+                    //loop through the includes and set the namespace alias 
+                    if (reference.Includes != null)
+                    {
+                        foreach (IEdmInclude include in reference.Includes)
+                        {
+                            if (include.Alias != null)
+                            {
+                                model.SetNamespaceAlias(include.Namespace, include.Alias);
+                            }
+                        }
+                    }
+                }
+
+                foreach (IEdmReference tmp in references)
+                {
+                    visitor = new EdmModelReferenceElementsXmlVisitor(this.model, this.writer, this.edmxVersion);

```

Comment:
Is there a reason to have two loops here? I think that the body of this loop can be incorporated into the body of the first loop, but I could be missing something

Reply:
@corranrogue9  the purpose of the two loops is to first have all the namespace-alias mapping for all the references before we start writing the references. 

For example 
`xml
<?xml version="1.0" encoding="utf-16"?>
<Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
	<Reference Uri="https://example.com/Org.OData.Authorization.V1.xml">
		<Include Namespace="Org.OData.Authorization.V1" Alias="Auth">
			<Include Namespace="Org.OData.Authorization.V1" Alias="Auth">
				<Annotation Term="Org.OData.Core.V1.LongDescription" String="Include Description." />
			</Include>
			<IncludeAnnotations TermNamespace="org.example.validation" />
			<IncludeAnnotations TermNamespace="org.example.display" Qualifier="Tablet" />
		</Include>
		<Reference Uri="https://example.com/Org.OData.Core.V1.xml">
			<Include Namespace="Org.OData.Core.V1" Alias="Core" />
			<IncludeAnnotations TermNamespace="org.example.hcm" TargetNamespace="com.example.Sales" />
			<IncludeAnnotations TermNamespace="org.example.hcm" Qualifier="Tablet" TargetNamespace="com.example.Person" />
			<Annotation Term="Core.LongDescription" String="EdmReference Description." />
		</Reference>
		<DataServices />
	</Reference>
</Edmx>
if we have such  a csdl, The <code>Org.OData.Core.V1 </code annotation is defined in the second reference but is used within the first reference. In that case, a look-up of the alias for that namespace won't be found.